src/charm.py

```diff
@@ -1146,23 +1167,279 @@ def _addon_reconciliation(self, addon_type):
                 raise exceptions.WordPressBlockedStatusException(
                     f"failed to uninstall {addon_type} {repr(addon)}"
                 )
+
     def _theme_reconciliation(self):
         """Reconciliation process for WordPress themes
 
         Install and uninstall themes to match the themes setting in config
         """
         self._addon_reconciliation("theme")
 
+    def _wp_option_update(self, option, value, format="plaintext"):
+        """Create or update a WordPress option value
+
+        If the option does not exist, wp option update will create one.
+        """
+        return self._wrapped_run_wp_cli(
+            ["wp", "option", "update", option, value, f"--format={format}"]
+        )
+
+    def _wp_option_delete(self, option):
+        """Delete a WordPress option
+
+        It's not an error to delete a non-existent option (it's a warning though)
+        """
+        return self._wrapped_run_wp_cli(["wp", "option", "delete", option])
+
+    def _wp_plugin_activate(self, plugin):
+        """Active a WordPress plugin"""
+        logger.info("activate plugin %s", repr(plugin))
+        return self._wrapped_run_wp_cli(["wp", "plugin", "activate", plugin])
+
+    def _wp_plugin_deactivate(self, plugin):
+        """Deactivate a WordPress plugin"""
+        logger.info("deactivate plugin %s", repr(plugin))
+        return self._wrapped_run_wp_cli(["wp", "plugin", "deactivate", plugin])
+
+    def _perform_plugin_activate_or_deactivate(self, plugin, action):
+        """Activate a WordPress plugin or deactivate a WordPress plugin.
+
+        It's not an error to activate an active plugin or deactivate an inactive plugin.
+        """
+        if action not in ("activate", "deactivate"):
+            raise ValueError(
+                f"Unknown activation_status {repr(action)}, "
+                "accept (activate, deactivate)"
+            )
+        current_plugins = self._wp_addon_list("plugin")
+        if not current_plugins.success:
+            return self._ExecResult(
+                success=False,
+                result=None,
+                message=f"failed to list installed plugins while {action} plugin {plugin}"
+            )
+        current_plugins = current_plugins.result
+        current_plugins_activate_status = {
+            p["name"]: p["status"] for p in current_plugins
+        }
+        if plugin not in current_plugins_activate_status:
+            return self._ExecResult(
+                success=False,
+                result=None,
+                message=f"{action} a non-existent plugin {plugin}"
+            )
+        is_active = current_plugins_activate_status[plugin] == "active"
```

Comment:
Do you think it is worthwhile to add a few new lines in this function to indicate a new set of logic? might make this more readable